### DISCUSSION TITLE
Is it possible to support interception of listing directory functionality?

### DISCUSSION THREAD

**Discussion by wzxinchen** [gitpod-io/openvscode-server#355] [General] [answered]
I don't know how to succinctly describe this feature, so I didn't search ahead.

I want to make a free WebIDE for end users (non-enterprise) based on OpenVscode.

This will involve permission issues. How to isolate the code of different users from each other?

Currently I have an idea of running a container for each user, so that they are naturally isolated.

But this is too expensive for hardware, after all, if I have a hundred users online at the same time, I have to run a hundred containers.

So I thought of another way to always run only one container, and the code of all users is stored inside the container, divided by folders.

This will face another problem, VSCode allows any user to open any folder, so how to safely isolate user data becomes a problem.

What I can think of is to intercept the process of listing directories of VSCode, and only list the directories that the current user is allowed to view.

**Comment by jeanp413** [comment, marked as answer]
Not sure how to help you regarding this, but your idea sounds a lot like Gitpod https://github.com/gitpod-io/gitpod
If you only want to deploy the frontend (the editor and workbench) without the backend (i.e. no terminal or compute power) like github.dev or vscode.dev, you can build only the frontend with `yarn gulp vscode-web-min`

**Reply by wzxinchen** [reply]
> Not sure how to help you regarding this, but your idea sounds a lot like Gitpod https://github.com/gitpod-io/gitpod If you only want to deploy the frontend (the editor and workbench) without the backend (i.e. no terminal or compute power) like github.dev or vscode.dev, you can build only the frontend with `yarn gulp vscode-web-min`

I tried code-server.sh and had some problems.

1. The problem of permissions, I should need to develop a plug-in to achieve, the plug-in receives a token, and then automatically selects a folder for the user, and locks only this folder. Is there any documentation to refer to if I want to implement this feature?

2. Questions for smart tips. It seems that intellisense for nuget packages is not supported? Now the SDK packages and the classes I wrote myself have intelligent tips.

**Reply by wzxinchen** [reply]
I thought about more details about permissions, maybe starting a docker per user would be a good choice
